### PR TITLE
Document error codes

### PR DESCRIPTION
**Pull request by DoodleBobBuffPants**

### CODE REVIEW

**Comment by Lyeeedar** [comment]
I think we should definitely have why an error might occur and advice on how to solve the error code in the docs. e.g. FAILED_REPEATABLE_MIGRATION - caused by a previous repeatable migration failing to execute - run flyway repair to remove failed entry.

**Comment by DoodleBobBuffPants** [comment]
> I think we should definitely have why an error might occur and advice on how to solve the error code in the docs. e.g. FAILED_REPEATABLE_MIGRATION - caused by a previous repeatable migration failing to execute - run flyway repair to remove failed entry.

I was thinking about structuring each one like:

```
- Error code
    - Cause
    - Solution
```

File: documentation/usage/errorcodes/index.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+## Validate Error Codes
+- `SCHEMA_DOES_NOT_EXIST` - The schema being validated against does not exist
+- `FAILED_REPEATABLE_MIGRATION` - A failed repeatable migration was detected
+- `FAILED_VERSIONED_MIGRATION` - A failed versioned migration was detected
+- `APPLIED_REPEATABLE_MIGRATION_NOT_RESOLVED` - A repeatable migration that was applied wasn't resolved in any supplied locations
+- `APPLIED_VERSIONED_MIGRATION_NOT_RESOLVED` - A versioned migration that was applied wasn't resolved in any supplied locations
+- `RESOLVED_REPEATABLE_MIGRATION_NOT_APPLIED` - A repeatable migration that was resolved has not been applied
+- `RESOLVED_VERSIONED_MIGRATION_NOT_APPLIED` - A versioned migration that was resolved has not been applied
+- `OUTDATED_REPEATABLE_MIGRATION` - An applied repeatable migration was resolved with a newer checksum and can be reapplied
+- `TYPE_MISMATCH` - The type of the resolved migration (`BASELINE`, `SQL`, `UNDO_SQL`, ...) is different from the applied migration's
```
Comment (MikielAgutu) [inline documentation/usage/errorcodes/index.md]: No need for an apostrophe to pluralize a noun.

**Review by MikielAgutu — CHANGES_REQUESTED** [review]
We should explain that the Error Codes are designed primarily for users of machine readable output. We should include explanations, along with examples of usage from the API and JSON output.

We should also explain the difference between `FAULT` and the other error codes. Being that `FAULT` indicates an unexpected error within Flyway (like a null ref) that isn't due to user configuration problems. In that case a user should contact support/create a GitHub issue.

This page is a good example I think: https://mariadb.com/kb/en/mariadb-error-codes/

File: documentation/usage/errorcodes/index.md (hunk 1 above)
Comment (DoodleBobBuffPants) [reply documentation/usage/errorcodes/index.md]: It's not pluralizing, it's ownership, I'm talking about the applied migration's type

**Review by Lyeeedar — APPROVED** [review]

File: documentation/usage/errorcodes/index.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+When Flyway commands fail, they throw an exception with a message to help you identify the problem. They also contain an error code which users of the API or those who have enabled machine readable output can inspect and handle accordingly. Below are details of each error code under the command that causes it along with a suggested solution.
+
+
+## Validate Error Codes
+These error codes are surfaced when running `validate` or `validateWithResult`.
+
+### `SCHEMA_DOES_NOT_EXIST`
+- **Caused by:** The schema being validated against does not exist
+- **Solution:** Manually create the schema or enable `flyway.createSchemas`
```
Comment (Lyeeedar) [inline documentation/usage/errorcodes/index.md]: Could we turn all these references to config parameters into links to the relevant docs pages?

**Review by Lyeeedar — CHANGES_REQUESTED** [review]

File: documentation/usage/errorcodes/index.md (hunk 2 above)
Comment (DoodleBobBuffPants) [reply documentation/usage/errorcodes/index.md]: See latest commit

**Review by Lyeeedar — APPROVED** [review]

File: documentation/usage/errorcodes/index.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+### `DESCRIPTION_MISMATCH`
+- **Caused by:** The description of the resolved migration is different from the applied migration's
+- **Solution:** Either revert the changes to the migration or run `repair` to update the schema history
+
+
+## General Error Codes
+These error codes may appear from any command, and are indicative of more general errors.
+
+### `FAULT`
```
Comment (MikielAgutu) [inline documentation/usage/errorcodes/index.md]: I think this should be moved up to the top. `VALIDATE_ERROR` can live under the 'validate error codes' heading.

**Pull request merged**